src/platform/plugins/shared/discover/public/application/main/state_management/redux/actions/tabs.ts

```diff
@@ -210,28 +210,35 @@ export const updateTabs: InternalStateThunkActionCreator<
       return tab;
     });
 
-    if (selectedItem?.id !== currentTabId) {
+    const selectedTab = selectedItem ?? currentTab;
+
+    // Push the selected tab ID to the URL, which creates a new browser history entry.
+    // This must be done before setting other URL state, which replace the history entry
+    // in order to avoid creating multiple browser history entries when switching tabs.
+    await tabsStorageManager.pushSelectedTabIdToUrl(selectedTab.id);
+
```

Comment:
After saving a discover session, `_tab` gets lost from the URL. Created a separate issue for it https://github.com/elastic/kibana/issues/236882